### PR TITLE
Remove the first bunch of BC layers

### PR DESCRIPTION
**Pull request by leofeyer**
These are just the low hanging fruits. There will be a lot more. 🙈

### CODE REVIEW

File: core-bundle/src/Command/UserListCommand.php [hunk 1]
```diff
@@ -61,10 +61,6 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $columns = $input->getOption('column');
 
         switch ($input->getOption('format')) {
-            case 'text':
-                trigger_deprecation('contao/core-bundle', '4.13', 'Using --format=text is deprecated and will be removed in Contao 5. Use --format=txt instead.');
```
Comment (m-vo) [inline core-bundle/src/Command/UserListCommand.php:65]: The deprecation message is probably wrong, is it? You should use `--format=text` instead of `--format=text`? :upside_down_face:

Comment (leofeyer) [reply core-bundle/src/Command/UserListCommand.php:65]: `--format=txt`

Comment (aschempp) [reply core-bundle/src/Command/UserListCommand.php:65]: seems correct to me (`format=txt`)

Comment (m-vo) [reply core-bundle/src/Command/UserListCommand.php:65]: Oh my… Sorry. :facepalm: I read over this at least 10 times and finally I found the difference. :eyeglasses:

**Review by m-vo — DISMISSED** [review]
:heart:

File: core-bundle/src/Controller/ContentElement/AbstractContentElementController.php [hunk 2]
```diff
@@ -31,14 +31,7 @@ public function __invoke(Request $request, ContentModel $model, string $section,
...
 
-        $response = $this->getResponse($template, $model, $request);
-
-        if (null === $response) {
-            trigger_deprecation('contao/core-bundle', '4.12', 'Returning null in %s::getResponse() is deprecated, return a Response instead.', static::class);
-            $response = $template->getResponse();
-        }
-
-        return $response;
+        return $this->getResponse($template, $model, $request);
```
Comment (aschempp) [inline core-bundle/src/Controller/ContentElement/AbstractContentElementController.php:34]: You also need to change the return type of the `getResponse` method

File: core-bundle/src/Controller/FrontendModule/AbstractFrontendModuleController.php [hunk 3]
```diff
@@ -37,14 +37,7 @@ public function __invoke(Request $request, ModuleModel $model, string $section,
...
 
-        $response = $this->getResponse($template, $model, $request);
-
-        if (null === $response) {
-            trigger_deprecation('contao/core-bundle', '4.12', 'Returning null in %s::getResponse() is deprecated, return a Response instead.', static::class);
-            $response = $template->getResponse();
-        }
-
-        return $response;
+        return $this->getResponse($template, $model, $request);
```
Comment (aschempp) [inline core-bundle/src/Controller/FrontendModule/AbstractFrontendModuleController.php:40]: You also need to change the return type of the `getResponse` method

File: core-bundle/src/Resources/contao/elements/ContentDownloads.php [hunk 4]
```diff
@@ -307,11 +307,6 @@ protected function compile()
 				array_multisort($files, SORT_NUMERIC, $auxDate, SORT_DESC);
 				break;
 
-			// Deprecated since Contao 4.0, to be removed in Contao 5.0
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/elements/ContentDownloads.php:310]: you also need to handle other `meta.txt` handling in `ContentDownloads`

File: core-bundle/src/Resources/contao/elements/ContentGallery.php [hunk 5]
```diff
@@ -162,11 +162,6 @@ protected function compile()
 				array_multisort($images, SORT_NUMERIC, $auxDate, SORT_DESC);
 				break;
 
-			// Deprecated since Contao 4.0, to be removed in Contao 5.0
-			case 'meta':
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/elements/ContentGallery.php:166]: you also need to handle other `meta.txt` handling in `ContentGallery`

File: core-bundle/src/Twig/Loader/TemplateLocator.php [hunk 6]
```diff
@@ -70,13 +69,7 @@ public function findThemeDirectories(): array
...
             }
 
-            try {
-                $slug = $this->themeNamespace->generateSlug(Path::makeRelative($themePath, 'templates'));
-            } catch (InvalidThemePathException $e) {
-                trigger_deprecation('contao/core-bundle', '4.12', 'Using a theme path with invalid characters has been deprecated and will throw an exception in Contao 5.0.');
-
-                continue;
-            }
+            $slug = $this->themeNamespace->generateSlug(Path::makeRelative($themePath, 'templates'));
```
Comment (aschempp) [inline core-bundle/src/Twig/Loader/TemplateLocator.php:75]: should we add a `@throws …` annotation for this?

File: manager-bundle/src/Routing/RouteLoader.php [hunk 7]
```diff
@@ -90,17 +90,6 @@ private function getConfigFile(): ?string
             }
         }
 
-        // Fallback to the legacy config file (see #566)
-        foreach (['routes.yaml', 'routes.yml', 'routing.yaml', 'routing.yml'] as $file) {
```
Comment (aschempp) [inline manager-bundle/src/Routing/RouteLoader.php:94]: We should also drop support for `'routing.yaml` and `routing.yml` in the `/config` folder (see a few lines above)

File: core-bundle/src/Resources/contao/modules/ModuleQuicknav.php [hunk 8]
```diff
@@ -130,13 +130,6 @@ protected function getQuicknavPages($pid, $level=1, $host=null)
 				$objSubpage->domain = $host;
 			}
 
-			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
-			if ($objSubpage->guests && !$objSubpage->protected && $isMember)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/ModuleQuicknav.php]: this should be a separate PR (including `tl_page.guests` and migration)

File: core-bundle/src/Resources/contao/modules/ModuleQuicklink.php [hunk 9]
```diff
@@ -91,13 +91,6 @@ protected function compile()
 		{
 			$objSubpage->loadDetails();
 
-			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
-			if ($objSubpage->guests && !$objSubpage->protected && $isMember)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/ModuleQuicklink.php]: this should be a separate PR (including `tl_page.guests` and migration)

File: core-bundle/src/Resources/contao/modules/ModuleCustomnav.php [hunk 10]
```diff
@@ -94,13 +94,6 @@ protected function compile()
 		{
 			$objModel->loadDetails();
 
-			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
-			if ($objModel->guests && !$objModel->protected && $isMember)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/ModuleCustomnav.php]: this should be a separate PR (including `tl_page.guests` and migration)

File: core-bundle/src/Resources/contao/modules/ModuleBooknav.php [hunk 11]
```diff
@@ -201,13 +201,6 @@ protected function getBookPages($intParentId, $groups, $time)
 		{
 			$objPage->loadDetails();
 
-			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
-			if ($objPage->guests && !$objPage->protected && $isMember)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/ModuleBooknav.php]: this should be a separate PR (including `tl_page.guests` and migration)

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 12]
```diff
@@ -322,13 +322,6 @@ protected function renderNavigation($pid, $level=1, $host=null, $language=null)
 				trigger_deprecation('contao/core-bundle', '4.12', 'Using a tabindex value greater than 0 has been deprecated and will no longer work in Contao 5.0.');
 			}
 
-			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
-			if ($objSubpage->guests && !$objSubpage->protected && $isMember)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/Module.php]: this should be a separate PR (including `tl_page.guests` and migration)

File: core-bundle/src/Controller/BackendController.php [hunk 13]
```diff
@@ -108,23 +106,6 @@ public function confirmAction(): Response
         return $controller->run();
     }
 
-    /**
-     * @Route("/file", name="contao_backend_file")
-     *
-     * @deprecated Deprecated since Contao 4.13, to be removed in Contao 5.0.
-     *             Use the picker instead.
-     */
-    public function fileAction(): Response
```
Comment (aschempp) [inline core-bundle/src/Controller/BackendController.php]: this should be reverted and is already in another PR

File: core-bundle/src/Controller/BackendController.php [hunk 14]
```diff
@@ -137,23 +118,6 @@ public function helpAction(): Response
         return $controller->run();
     }
 
-    /**
-     * @Route("/page", name="contao_backend_page")
-     *
-     * @deprecated Deprecated since Contao 4.13, to be removed in Contao 5.0.
-     *             Use the picker instead.
-     */
-    public function pageAction(): Response
```
Comment (aschempp) [inline core-bundle/src/Controller/BackendController.php]: this should be reverted and is already in another PR

File: core-bundle/src/Controller/ContentElement/AbstractContentElementController.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/Controller/ContentElement/AbstractContentElementController.php:34]: Changed in 7d922a1969d1b466a09e507902a70e065794a50a.

File: core-bundle/src/Controller/FrontendModule/AbstractFrontendModuleController.php (hunk 3 above)
Comment (leofeyer) [reply core-bundle/src/Controller/FrontendModule/AbstractFrontendModuleController.php:40]: Changed in 7d922a1969d1b466a09e507902a70e065794a50a.

File: core-bundle/src/Resources/contao/elements/ContentDownloads.php (hunk 4 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/elements/ContentDownloads.php:310]: Removed in 4c28151961d07efecd0a6c5b118f433b3183f550.

File: core-bundle/src/Resources/contao/elements/ContentGallery.php (hunk 5 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/elements/ContentGallery.php:166]: Removed in 4c28151961d07efecd0a6c5b118f433b3183f550.

File: manager-bundle/src/Routing/RouteLoader.php (hunk 7 above)
Comment (leofeyer) [reply manager-bundle/src/Routing/RouteLoader.php:94]: Changed in 8362dde51ebd32fa7162576085629d52a8365320.

File: core-bundle/src/Resources/contao/modules/ModuleQuicknav.php (hunk 8 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/ModuleQuicknav.php]: Reverted in 965ab6aa605345f3e860857032cebdeb8a3bdd8f.

File: core-bundle/src/Resources/contao/modules/ModuleQuicklink.php (hunk 9 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/ModuleQuicklink.php]: Reverted in 965ab6aa605345f3e860857032cebdeb8a3bdd8f.

File: core-bundle/src/Resources/contao/modules/ModuleCustomnav.php (hunk 10 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/ModuleCustomnav.php]: Reverted in 965ab6aa605345f3e860857032cebdeb8a3bdd8f.

File: core-bundle/src/Resources/contao/modules/ModuleBooknav.php (hunk 11 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/ModuleBooknav.php]: Reverted in 965ab6aa605345f3e860857032cebdeb8a3bdd8f.

File: core-bundle/src/Resources/contao/modules/Module.php (hunk 12 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php]: Revert in 965ab6aa605345f3e860857032cebdeb8a3bdd8f.

File: core-bundle/src/Controller/BackendController.php (hunk 13 above)
Comment (leofeyer) [reply core-bundle/src/Controller/BackendController.php]: Reverted in 576e7b51a52d5c13133b6445fc85e491c1c91a0c.

File: core-bundle/src/Controller/BackendController.php (hunk 14 above)
Comment (leofeyer) [reply core-bundle/src/Controller/BackendController.php]: Reverted in 576e7b51a52d5c13133b6445fc85e491c1c91a0c.

File: core-bundle/src/Twig/Loader/TemplateLocator.php (hunk 6 above)
Comment (leofeyer) [reply core-bundle/src/Twig/Loader/TemplateLocator.php:75]: Added in 576e7b51a52d5c13133b6445fc85e491c1c91a0c.

**Review by aschempp — APPROVED** [review]
